index.js

```diff
@@ -81,72 +81,77 @@ Toolkit.run(async (tools) => {
     return;
   }
 
-  try {
-    const current = pkg.version.toString();
-    // set git user
-    await tools.runInWorkspace('git', [
-      'config',
-      'user.name',
-      `"${process.env.GITHUB_USER || 'Automated Version Bump'}"`,
-    ]);
-    await tools.runInWorkspace('git', [
-      'config',
-      'user.email',
-      `"${process.env.GITHUB_EMAIL || 'gh-action-bump-version@users.noreply.github.com'}"`,
-    ]);
-
-    let currentBranch = /refs\/[a-zA-Z]+\/(.*)/.exec(process.env.GITHUB_REF)[1];
-    let isPullRequest = false;
-    if (process.env.GITHUB_HEAD_REF) {
-      // Comes from a pull request
-      currentBranch = process.env.GITHUB_HEAD_REF;
-      isPullRequest = true;
-    }
-    if (process.env['INPUT_TARGET-BRANCH']) {
-      // We want to override the branch that we are pulling / pushing to
-      currentBranch = process.env['INPUT_TARGET-BRANCH'];
-    }
-    console.log('currentBranch:', currentBranch);
-    // do it in the current checked out github branch (DETACHED HEAD)
-    // important for further usage of the package.json version
-    await tools.runInWorkspace('npm', ['version', '--allow-same-version=true', '--git-tag-version=false', current]);
-    console.log('current:', current, '/', 'version:', version);
-    let newVersion = execSync(`npm version --git-tag-version=false ${version}`).toString().trim().replace(/^v/, '');
-    newVersion = `${process.env['INPUT_TAG-PREFIX']}${newVersion}`;
-    await tools.runInWorkspace('git', ['commit', '-a', '-m', commitMessage.replace(/{{version}}/g, newVersion)]);
-
-    // now go to the actual branch to perform the same versioning
-    if (isPullRequest) {
-      // First fetch to get updated local version of branch
-      await tools.runInWorkspace('git', ['fetch']);
-    }
-    await tools.runInWorkspace('git', ['checkout', currentBranch]);
-    await tools.runInWorkspace('npm', ['version', '--allow-same-version=true', '--git-tag-version=false', current]);
-    console.log('current:', current, '/', 'version:', version);
-    newVersion = execSync(`npm version --git-tag-version=false ${version}`).toString().trim().replace(/^v/, '');
-    newVersion = `${process.env['INPUT_TAG-PREFIX']}${newVersion}`;
-    console.log(`::set-output name=newTag::${newVersion}`);
+  let push = process.env.INPUT_PUSH
+
+  if (push === 'true') {
```

Comment:
Can you make an early return, instead of all this indented code block?

```
if (push !== 'true') {
  tools.exit.success('Version bumped but not pushed!');
  return;
}
```

And keep the separate try/catch, they are here for a reason ;)